drivers/gpu/drm/panel/panel-waveshare-dsi.c

```diff
@@ -340,9 +340,8 @@ static int ws_panel_probe(struct i2c_client *i2c,
 	 */
 	drm_panel_add(&ts->base);
 
-	ts->dsi->mode_flags = (MIPI_DSI_MODE_VIDEO |
-			   MIPI_DSI_MODE_VIDEO_SYNC_PULSE |
-			   MIPI_DSI_MODE_LPM);
+	ts->dsi->mode_flags =  MIPI_DSI_MODE_VIDEO_HSE | MIPI_DSI_MODE_VIDEO |
+			   MIPI_DSI_CLOCK_NON_CONTINUOUS;
```

Comment:
Thank you for your further additions.
Yes, the 7.9 inch panel works fine on the pi4 but not on the pi5.
After modifying the mode, it ran fine when tested on both pi4 and pi5.

Reply:
That's not a very encouraging answer - it reads like "we changed some stuff until it worked on Pi 5, but we don't understand why". If that's not the case, please explain further.

Reply:
I've just tried the 7.9" panel out on my Pi5 (actually on a 6.6.9 kernel, but that's irrelevant).

You're right that the existing driver results in no output on Pi5.
Swapping from `MIPI_DSI_MODE_VIDEO_SYNC_PULSE` to `MIPI_DSI_MODE_VIDEO_HSE`, or just removing `MIPI_DSI_MODE_VIDEO_SYNC_PULSE` (to get sync events) both seem to work. That probably means that the video timings don't quite allow both HSYNC_START and HSYNC_END packets to be put out the bus.

Switching to `MIPI_DSI_CLOCK_NON_CONTINUOUS` is not needed.

In passing I note that the refresh rate for the 7.9" panel appears to come out at 73.71Hz. I did my best to copy the timings from those selected by your binary blob drivers, but that seems an odd refresh rate.

TBH I'm grateful for your input in actually making this driver do what is required. It would be nice to understand the full reasoning, but also happy enough if it works. I'm never intending to upstream this driver (others could as it has the relevant Signed-off-by tags), so it can be a little hacky if necessary.

Reply:
It's only for their panels, so adding the random MIPI_DSI_MODE_VIDEO_HSE and MIPI_DSI_CLOCK_NON_CONTINUOUS flags makes little difference to the world.
Likewise the odd refresh rate of 73.71Hz is quirky, but causes no harm anywhere else.

Merge away.

Reply:
I apologize for my delayed response, and I appreciate your understanding and tolerance.

The addition of MIPI_DSI_CLOCK_NON_CONTINUOUS was referenced from the panel-simple.c driver.

MIPI_DSI_MODE_VIDEO_SYNC_PULSE works well on the Raspberry Pi 4, but we are uncertain why it is not functioning on the Pi 5.

As for the refresh rate of 73.71Hz for the 7.9-inch panel, this specific refresh rate was determined during the development phase for debugging purposes. The stability of the display on the special panel is correlated with the main control clock frequency. It operates smoothly near the clock frequency of the Raspberry Pi's main control, ensuring that operations like sleep and wake do not encounter abnormal bugs. Therefore, at this frequency, the corresponding refresh rate is calculated as 73.71Hz.

We will continue to improve our documentation, and any additional discoveries will be updated and explained accordingly. Thank you once again.

Reply:
Thanks for the update.

Quirks like your 73.1Hz are almost inevitable, but it's nice to know that they are intentional rather than an oversight.

`MIPI_DSI_CLOCK_NON_CONTINUOUS` isn't necessarily wrong, it was just a change that seemed unnecessary.

I will try to find time to validate `MIPI_DSI_MODE_VIDEO_SYNC_PULSE` on Pi5 and ensure it is compliant with the DSI spec, but I'm expecting it to be. (I've been looking at it on other panels recently).
The flag is ignored on Pi4, so it won't have mattered whether the flag was incorrect previously.